src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerListener.java

```diff
@@ -29,6 +30,15 @@ public void onStarted(Run<?, ?> run, TaskListener listener) {
                     TaskListener taskListener = stepContext.get(TaskListener.class);
                     // encodeTo(Run) calls getDisplayName, which does not include the project name.
                     taskListener.getLogger().println("Starting building: " + ModelHyperlinkNote.encodeTo("/" + run.getUrl(), run.getFullDisplayName()));
+                    FlowNode parentNode = stepContext.get(FlowNode.class);
+                    if (parentNode != null) {
```

Comment:
@oleg-nenashev, I suppose that this code is not thread-safe as well. Is it correct?
I think to put `synchronized(this)` for `addAction/getAction`